target/linux/realtek/dts-5.10/rtl8393_panasonic_m48eg-pn28480k.dts

```diff
@@ -0,0 +1,352 @@
+
+#include "rtl839x.dtsi"
+#include "rtl83xx_panasonic_mxxeg-pn28xx0k.dtsi"
+
+#include <dt-bindings/interrupt-controller/irq.h>
+
+/ {
+	compatible = "panasonic,m48eg-pn28480k", "realtek,rtl8393-soc";
+	model = "Panasonic Switch-M48eG PN28480K";
+
+	aliases {
+		led-boot = &led_status_eco_green;
+		led-failsafe = &led_status_eco_amber;
+		led-running = &led_status_eco_green;
+		led-upgrade = &led_status_eco_green;
+	};
+
+	fan: gpio-fan {
+		compatible = "gpio-fan";
+		gpios = <&gpio0 19 GPIO_ACTIVE_HIGH>;
+		gpio-fan,speed-map = <6000 0
+				      8900 1>;
+		#cooling-cells = <2>;
+	};
+
+	/*
+	 * sfp0/1/2/3 are "combo" port with each TP port (45/46/47/48),
+	 * and they are connected to the RTL8218FB. Currently, there is
+	 * no support for the chip and only TP ports work by the RTL8218B
+	 * support.
+	 */
+	sfp0: sfp-p45 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c0>;
+		tx-fault-gpio = <&gpio1 0 GPIO_ACTIVE_HIGH>;
+		tx-disable-gpio = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio1 2 GPIO_ACTIVE_LOW>;
+		los-gpio = <&gpio1 3 GPIO_ACTIVE_HIGH>;
+	};
+
+	sfp1: sfp-p46 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1>;
+		tx-fault-gpio = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+		tx-disable-gpio = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio1 6 GPIO_ACTIVE_LOW>;
+		los-gpio = <&gpio1 7 GPIO_ACTIVE_HIGH>;
+	};
+
+	sfp2: sfp-p47 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c2>;
+		tx-fault-gpio = <&gpio1 8 GPIO_ACTIVE_HIGH>;
+		tx-disable-gpio = <&gpio1 9 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio1 10 GPIO_ACTIVE_LOW>;
+		los-gpio = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+	};
+
+	sfp3: sfp-p48 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c3>;
+		tx-fault-gpio = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+		tx-disable-gpio = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio1 14 GPIO_ACTIVE_LOW>;
+		los-gpio = <&gpio1 15 GPIO_ACTIVE_HIGH>;
+	};
+
+	thermal-zones {
+		cpu-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <2000>;
+
+			thermal-sensors = <&tsens0>;
+
+			trips {
+				cpu_alert: trip-point {
+					temperature = <45000>;
+					hysteresis = <4000>;
+					type = "active";
+				};
+
+				cpu_crit {
+					temperature = <55000>;
+					hysteresis = <1000>;
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map {
+					trip = <&cpu_alert>;
+					cooling-device = <&fan 0 1>;
+				};
+			};
+		};
+
+		/*
+		 * Official Spcifications about Temperature (System)
+		 *
+		 * Fan Speed setting:
+		 * - Low : 0-40 celsius
+		 * - High: 0-50 celsius
+		 */
+		sys-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <2000>;
+
+			thermal-sensors = <&tsens1>;
+
+			trips {
+				sys_alert: trip-point {
+					temperature = <40000>;
+					hysteresis = <4000>;
+					type = "active";
+				};
+
+				sys_crit {
+					temperature = <50000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map {
+					trip = <&sys_alert>;
+					cooling-device = <&fan 0 1>;
+				};
```

Comment:
How does the stock firmware deal with the two fans? Are they also run independently, or do they turn on/off together? I'm wondering how much sense it makes, physically, to consider this as two independent thermal zones if the fans are actually mounted in the chassis, and not e.g. on the CPU.

Reply:
2x fans of Switch-M48eG PN28480K are controlled integrally like 1x fan.
The thermal zone of Linux Kernel supports only one sensor per zone, so 2x zones are required for 2x sensors, SoC and system. Both zones controls the same fan unit.

I checked behavior on the device (set thresholds soc->37, sys->36 for test):

- SoC: 35.5 celsius, Sys: 32.5 celsius ---> speed: Low

- SoC: 37.5 celsius, Sys: 34.5 celsius ---> speed: High

- SoC: 40.5 celsius, Sys: 36.0 celsius ---> speed: High

- SoC: 37.0 celsius, Sys: 33.5 celsius ---> speed: High

- SoC: 36.5 celsius, Sys: 33.0 celsius ---> speed: Low

Reply:
Right, there is only one fan section. My mind must have interpreted the two speeds as two fans... My apologies!

All very good then. Having the both zones independently enable the fans if either gets too hot sounds exactly like what one would want.